### ISSUE TITLE
health check fails with java.lang.IllegalArgumentException: port out of range:-1

### ISSUE THREAD

**Issue by vsevel** [quarkiverse/quarkus-artemis#580]
when I call `/q/health` on an application running the jca adapter, and connected **successfully** to a hornetq cluster, I get the following exception:
```
2024-07-31 12:02:44,006 INFO  [org.acm.MyProducer] (executor-thread-1) sending jms message Wed Jul 31 12:02:44 CEST 2024 from host=x
2024-07-31 12:02:49,008 INFO  [org.acm.MyProducer] (executor-thread-1) sending jms message Wed Jul 31 12:02:49 CEST 2024 from host=x
2024-07-31 12:02:54,016 INFO  [org.acm.MyProducer] (executor-thread-2) sending jms message Wed Jul 31 12:02:54 CEST 2024 from host=x

...
2024-07-31 12:02:56,066 WARN  [org.apa.act.art.cor.client] (executor-thread-1) AMQ212007: connector.create or connectorFactory.createConnector should never throw an exception, implementation is badly behaved, but we will deal with it anyway.: java.lang.IllegalArgumentException: port out of range:-1
        at java.base/java.net.InetSocketAddress.checkPort(InetSocketAddress.java:153)
        at java.base/java.net.InetSocketAddress.<init>(InetSocketAddress.java:234)
        at org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnector.createConnection(NettyConnector.java:885)
        at org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnector.createConnection(NettyConnector.java:877)
        at org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnector.createConnection(NettyConnector.java:859)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.openTransportConnection(ClientSessionFactoryImpl.java:1222)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.createTransportConnection(ClientSessionFactoryImpl.java:1343)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.createTransportConnection(ClientSessionFactoryImpl.java:1263)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.establishNewConnection(ClientSessionFactoryImpl.java:1506)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.getConnection(ClientSessionFactoryImpl.java:1084)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.getConnectionWithRetry(ClientSessionFactoryImpl.java:969)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.connect(ClientSessionFactoryImpl.java:287)
        at org.apache.activemq.artemis.core.client.impl.ClientSessionFactoryImpl.connect(ClientSessionFactoryImpl.java:303)
        at org.apache.activemq.artemis.core.client.impl.ServerLocatorImpl.createSessionFactory(ServerLocatorImpl.java:718)
        at org.apache.activemq.artemis.api.core.client.ServerLocator_JD5VznKFNQ2EhTb-yyf2Ep9BRq8_Synthetic_ClientProxy.createSessionFactory(Unknown Source)
        at io.quarkus.artemis.core.runtime.health.ServerLocatorHealthCheck.call(ServerLocatorHealthCheck.java:79)
        at io.quarkus.artemis.core.runtime.health.ServerLocatorHealthCheck_ClientProxy.call(Unknown Source)
        at io.smallrye.context.impl.wrappers.SlowContextualSupplier.get(SlowContextualSupplier.java:21)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromItemSupplier.subscribe(UniCreateFromItemSupplier.java:28)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)

2024-07-31 12:02:56,091 INFO  [io.sma.health] (vert.x-eventloop-thread-0) SRHCK01001: Reporting health down status: {"status":"DOWN","checks":[{"name":"Artemis Core health check","status":"DOWN","data":{"<default>":"DOWN"}}]}

```

I am using the same reproducer described in https://github.com/quarkiverse/quarkus-artemis/issues/567, except I added:
```
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-smallrye-health</artifactId>
    </dependency>
```

when I look at the `serverLocators` in `ServerLocatorHealthCheck` I see:
```
<default> -> {ServerLocator_JD5VznKFNQ2EhTb-yyf2Ep9BRq8_Synthetic_ClientProxy@5708} "ServerLocatorImpl [initialConnectors=[TransportConfiguration(name=null, factory=org-apache-activemq-artemis-core-remoting-impl-netty-NettyConnectorFactory)?port=-1&host=null], discoveryGroupConfiguration=null]"
```

I am just wondering if this health check supports connections created from the jca adapter.
cc @zhfeng @gastaldi

**Comment by gastaldi** [comment]
It seems that it's missing the connection parameters? It's weird that the host and port are null in this case. Maybe @turing85 has an idea?

**Comment by vsevel** [comment]
may be a stupid question, but when using the jca adaptor, are `ServerLocator` proper cdi beans, that get properly discovered by the processor?

**Comment by gastaldi** [comment]
@vsevel that makes sense, does it work if you don't use the JCA adaptor?

**Comment by zhfeng** [comment]
I'm checking ...

**Comment by zhfeng** [comment]
Hmm, it looks like we need to implement such a `ResourceAdapterHealthCheck` in `ra` like `ServerLocatorHealthCheck` in `core` and `ConnectionFactoryHealthCheck` ? @turing85

**Comment by zhfeng** [comment]
Or @gastaldi we should add `HealthCheck` support in `quarkus-ironjacamar`?

**Comment by gastaldi** [comment]
@turing85 AFAIK the `ArtemisResourceAdapterFactory` doesn't register any `ServerLocator`s in ArC, perhaps that's what @zhfeng means?

**Comment by turing85** [comment]
Then I do not understand the stack trace. It points to:
```
at io.quarkus.artemis.core.runtime.health.ServerLocatorHealthCheck.call(ServerLocatorHealthCheck.java:79)
```
aka:

https://github.com/quarkiverse/quarkus-artemis/blob/bbd611558c18b742d80362ad77c7e20f6c889fe9/core/runtime/src/main/java/io/quarkus/artemis/core/runtime/health/ServerLocatorHealthCheck.java#L74-L86

**Comment by gastaldi** [comment]
It seems to be produced [here](https://github.com/quarkiverse/quarkus-artemis/blob/main/core/deployment/src/main/java/io/quarkus/artemis/core/deployment/ArtemisCoreProcessor.java#L131)?

**Comment by zhfeng** [comment]
@turing85 see https://github.com/quarkiverse/quarkus-artemis/blob/49d6f320bdad9b0421df7f3bbefeecc1fe7e592a/core/deployment/src/main/java/io/quarkus/artemis/core/deployment/ArtemisCoreProcessor.java#L103-L115

because we did not produce `ArtemisJmsBuildItem` in `quarkus-artemis-ra`, the ServerLocator is generated. Maybe  we need to add such similar codes in `quarkus-artemis-ra`

```java
    @BuildStep
    void load(BuildProducer<ArtemisJmsBuildItem> artemisJms) {
        artemisJms.produce(new ArtemisJmsBuildItem());
    }
```

**Comment by zhfeng** [comment]
Sorry, it looks like I create `ServerLocator` from the following
https://github.com/quarkiverse/quarkus-artemis/blob/0a67d2c8ea4bd943e0ea10b9bdae2c5491585718/ra/runtime/src/main/java/io/quarkus/artemis/jms/ra/ArtemisResourceAdapterRecorder.java#L17-L29

which does not support the cluster configuration?

**Comment by zhfeng** [comment]
Hi @gastaldi ,

I have a proposal to introduce a new `ConnectionFactoryHealthCheck` which is similar with `quarkus-artemis-jms`. It needs to check the `IronJacamarBuildtimeConfig` for the artemis config.  See the following

```java
private void processKnownBeans(IronJacamarBuildtimeConfig buildTimeConfigs) {
        for (String name : buildTimeConfigs.resourceAdapters().keySet()) {
            if (buildTimeConfigs.resourceAdapters().get(name).ra().kind().equals("artemis")) {
                Annotation identifier;
                if (ArtemisUtil.isDefault(name)) {
                    identifier = Default.Literal.INSTANCE;
                } else {
                    identifier = Identifier.Literal.of(name);
                }
                ConnectionFactory connectionFactory;
                try (InstanceHandle<ConnectionFactory> handle = Arc.container().instance(ConnectionFactory.class, identifier)) {
                    connectionFactory = handle.get();
                }
                if (connectionFactory != null) {
                    connectionFactories.put(name, connectionFactory);
                }
            }
        }
    }
``` 

But currently `IronJacamarBuildtimeConfig` is `ConfigPhase.BUILD_TIME`, so it can not be access at runtime. I want to change it to `ConfigPhase.BUILD_AND_RUN_TIME_FIXED`. Is it good for you?

**Comment by zhfeng** [comment]
@gastaldi that might be a little bit complicated. It needs to convert `ResourceAdapterKindBuildItem` and pass it to a runtime value which should be injected in `ConnectionFactoryHealthCheck`. Do you see another simple way?

**Comment by gastaldi** [comment]
Not sure why you need to store connection factories in a Map when you could have a runtime CDI bean doing: 

```
@Inject 
Instance<ConnectionFactory> connectionFactories
```

And looking the connection factory using: `connectionFactories.select(Identifier.Literal.of(name)).get()`?

**Comment by gastaldi** [comment]
Anyway, I merged the PR you provided, it looks good. Let me know if we can proceed with the release in quarkus-ironjacamar

**Comment by turing85** [comment]
> Not sure why you need to store connection factories in a Map when you could have a runtime CDI bean doing:
> 
> ```
> @Inject 
> Instance<ConnectionFactory> connectionFactories
> ```
> 
> And looking the connection factory using: `connectionFactories.select(Identifier.Literal.of(name)).get()`?

@gastaldi [I tookt that personally](https://github.com/quarkiverse/quarkus-artemis/issues/580#issuecomment-2263266646).

**Comment by turing85** [comment]
The fix has been released with versions [`3.4.1`](https://github.com/quarkiverse/quarkus-artemis/releases/tag/3.4.1) and [`3.2.3`](https://github.com/quarkiverse/quarkus-artemis/releases/tag/3.2.3).
